### PR TITLE
[in_app_purchase_storekit] convert TranslatorTests to swift

### PR DESCRIPTION
**Pull request by LouiseHsu**
Part of https://github.com/flutter/flutter/issues/151624

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_storekit/example/macos/RunnerTests/TranslatorTests.swift [hunk 1]
```diff
@@ -0,0 +1 @@
+../../shared/RunnerTests/TranslatorTests.swift
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/macos/RunnerTests/TranslatorTests.swift:1]: what are these files

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.swift [hunk 2]
```diff
@@ -82,7 +82,7 @@ final class InAppPurchasePluginTests: XCTestCase {
       }
 
       XCTAssertEqual(unwrappedProducts.count, 1)
-      XCTAssertEqual(response.invalidProductIdentifiers, nil)
+      XCTAssertEqual(response.invalidProductIdentifiers, [])
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.swift:85]: why this behavior change?

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+// found in the LICENSE file.
+
+import Foundation
+import StoreKit
+import XCTest
+
+@testable import in_app_purchase_storekit
+
+final class ObjectTranslatorTest: XCTestCase {
+  var periodMap: [String: Any] {
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift:12]: these can be private

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+      "storefront": storefrontMap,
+      "transaction": transactionMap,
+    ]
+  }
+
+  func testSKProductSubscriptionPeriodStubToMap() {
+    let period = SKProductSubscriptionPeriodStub(map: periodMap)
+    let map = FIAObjectTranslator.getMapFrom(period)
+
+    XCTAssertEqual(map as NSDictionary, periodMap as NSDictionary)
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift:125]: Does this test pass? it should be `XCTAssertEqualObjects`

Comment (hellohuanlin) [reply packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift:125]: oh nevermind, this is swift.

**Review by hellohuanlin — APPROVED** [review]

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.swift (hunk 2 above)
Comment (LouiseHsu) [reply packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.swift:85]: obj-c let me pass in a nil with no objections, but invalidProductIdentifiers is not an optional field in swift version of the api https://developer.apple.com/documentation/storekit/skproductsresponse/1505985-invalidproductidentifiers

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift (hunk 3 above)
Comment (LouiseHsu) [reply packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/TranslatorTests.swift:12]: updated!

File: packages/in_app_purchase/in_app_purchase_storekit/example/macos/RunnerTests/TranslatorTests.swift (hunk 1 above)
Comment (LouiseHsu) [reply packages/in_app_purchase/in_app_purchase_storekit/example/macos/RunnerTests/TranslatorTests.swift:1]: the actual test file gets put in shared/RunnerTests and the ios/macos workspaces just symlink to it. i guess this is what the symlinks look like

**Pull request merged**